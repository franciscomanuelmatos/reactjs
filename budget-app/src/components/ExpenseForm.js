import React from 'react';
import { connect } from 'react-redux';

import { addExpense } from '../actions/expenses';

class ExpenseForm extends React.Component {
  state = {
    description: '',
    amount: '',
    note: ''
  }

  onDescriptionChange = (event) => {
    const description = event.target.value;
    this.setState(() => ({ description }));
  }

  onAmountChange = (event) => {
    const amount = event.target.value;
    if (amount.match(/^\d*(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  }

  onNoteChange = (event) => {
    const note = event.target.value;
    this.setState(() => ({ note }));
  }

  render() {
    return (
      <div>
        <form>
          <input 
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input 
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <textarea 
            placeholder="Add a note for your expense"
            value={this.state.note}
            onChange={this.onNoteChange}
          >
          </textarea>
          <button>Add Expense</button>
        </form>
      </div>
    )
  }
}

export default ExpenseForm;