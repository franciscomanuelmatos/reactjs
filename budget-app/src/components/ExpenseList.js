import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import ExpenseListFilters from './ExpenseListFilters';
import { getVisibleExpenses } from '../selectors/expenses';

const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    <ExpenseListFilters />

    {props.expenses.map((expense) => (
      <ExpenseListItem expense={expense} key={expense.id} /> 
    ))}
  </div>
);

const mapStateToProps = (state) => ({
  expenses: getVisibleExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);
