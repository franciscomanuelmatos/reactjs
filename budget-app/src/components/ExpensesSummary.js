import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import { getVisibleExpenses } from '../selectors/expenses';
import { getExpensesTotal } from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
  const expenseWord = props.expenseCount === 1 ? 'expense' : 'expenses';
  const total = numeral(props.expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      {props.expenseCount > 0 && 
        <h2>Viewing {props.expenseCount} {expenseWord} totalling {total} </h2>
      }
    </div>
  )
}
  
const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses)
  }
};

export default connect(mapStateToProps)(ExpensesSummary);