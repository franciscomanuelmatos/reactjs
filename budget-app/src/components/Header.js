import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <nav>
      <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
      <NavLink to="/create" activeClassName="is-active" exact={true}>Create expense</NavLink>
      <NavLink to="/edit" activeClassName="is-active" exact={true}>Edit expense</NavLink>
      <NavLink to="/help" activeClassName="is-active" exact={true}>Help</NavLink>
    </nav>
  </header>
);

export default Header;