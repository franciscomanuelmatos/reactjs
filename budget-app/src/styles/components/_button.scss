.button {
  background: $blue;
  border: none;
  border-radius: .5rem;
  color: white;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
}