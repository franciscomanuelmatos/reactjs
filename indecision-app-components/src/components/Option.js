import React from 'react';

const Option = (props) => {
    return (
        <div>
            {props.option}
            <button 
            onClick={(e) => props.handleDeleteOption(props.option)}>
            Delete
            </button>
        </div>
    );
}

export default Option;