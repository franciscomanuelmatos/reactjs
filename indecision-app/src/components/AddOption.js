import React from 'react';

export default class AddOption extends React.Component {
    state = {
        errorMessage: undefined
    }

    handleFormSubmission = (e) => {
        e.preventDefault();
        const newOption = e.target.elements.option.value.trim();
        const errorMessage = this.props.handleNewOption(newOption);
        
        if (!errorMessage) {
            e.target.elements.option.value = '';
        }
        
        this.setState(() => ({ errorMessage }));
        
    }

    render() {
        return (
            <div>
                {this.state.errorMessage && <p className="add-option-error">{this.state.errorMessage}</p>}
                <form className="add-option" onSubmit={this.handleFormSubmission}>
                    <input className="add-option__input" type="text" name="option" />
                    <button className="button">Add option</button>
                </form>
            </div>
        );
    }
}

