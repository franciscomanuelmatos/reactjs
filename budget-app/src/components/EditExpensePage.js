import React from 'react';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id,expense);
    this.props.history.push('/');
  }

  onClick = () => {
    this.props.startRemoveExpense(this.props.expense);
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <ExpenseForm
          expense = {this.props.expense}
          onSubmit = {this.onSubmit}
        />
        <button onClick={this.onClick}>Remove</button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  const id = props.match.params.id;

  return {
    expense: state.expenses.find((expense) => expense.id === id)
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    editExpense: (id, expense) => {
      dispatch(editExpense(id, expense));
    },
    startRemoveExpense: (expense) => { dispatch(startRemoveExpense(expense)) }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
