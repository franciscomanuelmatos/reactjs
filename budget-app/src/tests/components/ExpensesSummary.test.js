import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';
import configNumeral from '../../config/numeral';
import expenses from '../fixtures/expenses';
import { getExpensesTotal } from '../../selectors/expenses-total';

beforeAll(() => {
  configNumeral();
});

test('Should not render ExpensesSummary when there are no expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={0} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={expenses.length} expensesTotal={getExpensesTotal(expenses)} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary correctly with a single expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={getExpensesTotal([expenses[0]])} />);
  expect(wrapper).toMatchSnapshot();
});