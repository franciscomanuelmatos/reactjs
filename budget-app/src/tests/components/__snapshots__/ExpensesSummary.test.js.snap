// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not render ExpensesSummary when there are no expenses 1`] = `<div />`;

exports[`Should render ExpensesSummary correctly 1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expenses
     totalling 
    €1 141,95
     
  </h2>
</div>
`;

exports[`Should render ExpensesSummary correctly with a single expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    €1,95
     
  </h2>
</div>
`;
